Feat  support export transcription

TakalaWang opened the pull request:
https://github.com/user-attachments/assets/300905dc-5744-4e5d-8bab-c2b5645ce019


This pull request introduces new localization strings for export functionality, updates dependencies to support PDF generation and data export, and modifies the lock file to reflect these changes. Below is a breakdown of the most important changes:

### Localization Updates:
* Added new strings in `messages/en.json` and `messages/zh.json` to support export functionality, including titles, descriptions, and actions for selecting participants and groups for transcript export. [[1]](diffhunk://#diff-5d9e5048516ec2fe83ca06813e79c3b8b44c6c96294f25c7311db311e28eeaeeL452-R461) [[2]](diffhunk://#diff-b1da422c0458d9d7093c9418675ac11a7d276d2d366636a493cca8e90dd75912L465-R474)

### Dependency Additions:
* Updated `package.json` to include new dependencies: `html2canvas`, `jspdf`, `jspdf-autotable`, and `jszip`, which are essential for generating and exporting PDFs and handling compressed files. [[1]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R96-R99) [[2]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R42)

### Lock File Updates:
* Added multiple packages to `pnpm-lock.yaml` to support new functionality, including `html2canvas`, `jspdf`, `jspdf-autotable`, `jszip`, and their related dependencies like `rgbcolor`, `stackblur-canvas`, and `svg-pathdata`. [[1]](diffhunk://#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR13-R24) [[2]](diffhunk://#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR5705-R5724) [[3]](diffhunk://#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR5782-R5787)

JacobLinCool commented on src/lib/components/session/HostView.svelte:
```diff
@@ -516,6 +582,170 @@
+				<div style="margin-bottom: 20px;">
+					<p><strong>Session:</strong> ${sessionTitle}</p>
+					<p><strong>學生:</strong> ${userName}</p>
+					<p><strong>生成時間:</strong> ${new Date().toLocaleString('zh-TW')}</p>
```
生成時間 sounds strange, and no i18n here?

JacobLinCool commented on src/lib/components/session/HostView.svelte:
```diff
@@ -146,6 +204,12 @@
 					);
 					groups.set(groupsData);
+
+					// 存儲小組數據用於 PDF 生成
```
not useful

JacobLinCool commented:
I have 2 concerns:
1. The new UI has become too complex. I suggest adding a toggle (switch) to show or hide the selects and buttons, in order to prevent the initial UI from displaying too much information.
2. Making `HostView.svelte` larger feels unfortunate. Some logic should be extracted to improve maintainability.

TakalaWang commented:
Updated demo video:
https://github.com/user-attachments/assets/685cc188-882f-4cf4-85d5-a51ea25d5a89



This pull request introduces several changes to enhance the application's functionality and dependencies. Key updates include the addition of new language strings for transcript export features, updates to `package.json` to include libraries for PDF generation and file manipulation, and corresponding changes in the `pnpm-lock.yaml` file to lock versions of new dependencies.

### Language updates for transcript export functionality:
* [`messages/en.json`](diffhunk://#diff-5d9e5048516ec2fe83ca06813e79c3b8b44c6c96294f25c7311db311e28eeaeeL452-R494): Added strings for transcript export options, PDF generation statuses, and error messages related to exporting transcripts.
* [`messages/zh.json`](diffhunk://#diff-b1da422c0458d9d7093c9418675ac11a7d276d2d366636a493cca8e90dd75912L465-R507): Added corresponding Chinese translations for the new transcript export-related strings.

### Dependency additions:
* [`package.json`](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R42): Added dependencies for `jspdf`, `jspdf-autotable`, `jszip`, `html2canvas`, and `xlsx` to support PDF generation, table formatting, file compression, and Excel file handling. [[1]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R42) [[2]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R62-R67) [[3]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R89)

### Dependency version locking:
* [`pnpm-lock.yaml`](diffhunk://#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL13-L15): Locked versions for newly added dependencies, including `jspdf`, `jspdf-autotable`, `jszip`, `html2canvas`, and `xlsx`. Removed the previously locked `xlsx` dependency under `dependencies`. [[1]](diffhunk://#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL13-L15) [[2]](diffhunk://#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR71-R73) [[3]](diffhunk://#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR131-R148) [[4]](diffhunk://#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR212-R214)
